### PR TITLE
Mm 30361

### PR DESCRIPTION
**Pull request by furqanmlk**
#### Summary
Cypress/E2E: Automate backlogs - LDAP Settings (5 test cases)

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-30361

#### Screenshots

image

image

### CODE REVIEW

File: e2e/cypress/integration/enterprise/ldap/ldap_setting_spec.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+describe('LDAP settings', () => {
+    beforeEach(() => {
+        // Login as sysadmin
+        cy.apiAdminLogin();
+
+        // Check if server has license for LDAP
+        cy.apiRequireLicenseForFeature('LDAP');
+    });
+    it('MM-T2699 Connection test button - Successful', () => {
```
Comment (josephbaylon) [inline e2e/cypress/integration/enterprise/ldap/ldap_setting_spec.js:24]:
```suggestion

    it('MM-T2699 Connection test button - Successful', () => {
```

File: e2e/cypress/integration/enterprise/ldap/ldap_setting_spec.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        cy.visitLDAPSettings();
+
+        // click "AD/LDAP Test"
+        cy.findByRole('button', {name: /ad\/ldap test/i}).click();
+
+        // * Confirmation message saying the connection is successful.
+        cy.findByText(/ad\/ldap test successful/i).should('be.visible');
+        cy.findByTitle(/success icon/i).should('be.visible');
+    });
+    it('MM-T2700 LDAP username required', () => {
```
Comment (josephbaylon) [inline e2e/cypress/integration/enterprise/ldap/ldap_setting_spec.js:35]:
```suggestion

    it('MM-T2700 LDAP username required', () => {
```

File: e2e/cypress/integration/enterprise/ldap/ldap_setting_spec.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+        // * Verifying message "AD/LDAP field "Username Attribute" is required."
+        cy.findByText('AD/LDAP field "Username Attribute" is required.').should('be.visible');
+
+        // * Set back to what it was
+        cy.findByLabelText(/username attribute:/i).click().type('uid');
+        cy.findByText(/save/i).click();
+        cy.findByRole('button', {name: /save/i}).should('be.disabled');
+    });
+    it('MM-T2701 LDAP LoginidAttribute required', () => {
```
Comment (josephbaylon) [inline e2e/cypress/integration/enterprise/ldap/ldap_setting_spec.js:52]:
```suggestion

    it('MM-T2701 LDAP LoginidAttribute required', () => {
```

File: e2e/cypress/integration/enterprise/ldap/ldap_setting_spec.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        cy.visitLDAPSettings();
+
+        // # Try to save LDAP settings with blank Loginid
+        cy.get('[data-testid="LdapSettings.LoginIdAttributeinput"]').click().clear();
+        cy.findByText(/save/i).click();
+
+        // * Verifying Error Message
+        cy.findByText(/ad\/ldap field "login id attribute" is required./i).should('be.visible');
+    });
+    it('MM-T2704 Create new LDAP account from login page', () => {
```
Comment (josephbaylon) [inline e2e/cypress/integration/enterprise/ldap/ldap_setting_spec.js:62]:
```suggestion

    it('MM-T2704 Create new LDAP account from login page', () => {
```

File: e2e/cypress/integration/enterprise/ldap/ldap_setting_spec.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        cy.findByTitle(/success icon/i).should('be.visible');
+    });
+    it('MM-T2700 LDAP username required', () => {
+        cy.visitLDAPSettings();
+
+        // # Remove text from Username Attribute
+        cy.findByLabelText(/username attribute:/i).click().clear();
+
+        // # Click Save
+        cy.findByText(/save/i).click();
```
Comment (josephbaylon) [inline e2e/cypress/integration/enterprise/ldap/ldap_setting_spec.js:42]:
```suggestion
        cy.findByRole('button', {name: /save/i}).click();
```

File: e2e/cypress/integration/enterprise/ldap/ldap_setting_spec.js [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+        // * Verifying message "AD/LDAP field "Username Attribute" is required."
+        cy.findByText('AD/LDAP field "Username Attribute" is required.').should('be.visible');
+
+        // * Set back to what it was
+        cy.findByLabelText(/username attribute:/i).click().type('uid');
+        cy.findByText(/save/i).click();
```
Comment (josephbaylon) [inline e2e/cypress/integration/enterprise/ldap/ldap_setting_spec.js:49]:
```suggestion
        cy.findByRole('button', {name: /save/i}).click();
```

File: e2e/cypress/integration/enterprise/ldap/ldap_setting_spec.js [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+        cy.findByRole('button', {name: /save/i}).should('be.disabled');
+    });
+    it('MM-T2701 LDAP LoginidAttribute required', () => {
+        cy.visitLDAPSettings();
+
+        // # Try to save LDAP settings with blank Loginid
+        cy.get('[data-testid="LdapSettings.LoginIdAttributeinput"]').click().clear();
+        cy.findByText(/save/i).click();
```
Comment (josephbaylon) [inline e2e/cypress/integration/enterprise/ldap/ldap_setting_spec.js:57]:
```suggestion
        cy.findByRole('button', {name: /save/i}).click();
```

File: e2e/cypress/integration/enterprise/ldap/ldap_setting_spec.js [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+        // * Set back to what it was
+        cy.findByLabelText(/username attribute:/i).click().type('uid');
+        cy.findByText(/save/i).click();
+        cy.findByRole('button', {name: /save/i}).should('be.disabled');
+    });
+    it('MM-T2701 LDAP LoginidAttribute required', () => {
+        cy.visitLDAPSettings();
+
+        // # Try to save LDAP settings with blank Loginid
+        cy.get('[data-testid="LdapSettings.LoginIdAttributeinput"]').click().clear();
```
Comment (josephbaylon) [inline e2e/cypress/integration/enterprise/ldap/ldap_setting_spec.js:56]:
```suggestion
        cy.findByTestId('LdapSettings.LoginIdAttributeinput').click().clear();
```

File: e2e/cypress/support/ldap_commands.js [hunk 9]
```diff
@@ -7,6 +7,11 @@ Cypress.Commands.add('doLDAPExistingLogin', () => {
     cy.findByText('Click here to sign in.').should('be.visible').click();
 });
 
+Cypress.Commands.add('visitLDAPSettings', () => {
+    // # Go to LDAP settings Page
+    cy.visit('/admin_console/authentication/ldap');
```
Comment (josephbaylon) [inline e2e/cypress/support/ldap_commands.js:12]:
```suggestion
    cy.visit('/admin_console/authentication/ldap');
    cy.get('.admin-console__header').should('be.visible').and('have.text', 'AD/LDAP');
```

**Review by josephbaylon — CHANGES_REQUESTED** [review]
@furqanmlk Thanks!! Please apply minor suggestions below for consistency and stability.

File: e2e/cypress/integration/enterprise/ldap/ldap_setting_spec.js (hunk 5 above)
Comment (furqanmlk) [reply e2e/cypress/integration/enterprise/ldap/ldap_setting_spec.js]: ✅

File: e2e/cypress/integration/enterprise/ldap/ldap_setting_spec.js (hunk 7 above)
Comment (furqanmlk) [reply e2e/cypress/integration/enterprise/ldap/ldap_setting_spec.js:60]: ✅

Comment (furqanmlk) [reply e2e/cypress/integration/enterprise/ldap/ldap_setting_spec.js:60]: ✅

File: e2e/cypress/integration/enterprise/ldap/ldap_setting_spec.js [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+        cy.findByRole('button', {name: /save/i}).click();
+        cy.findByRole('button', {name: /save/i}).should('be.disabled');
+    });
+
+    it('MM-T2701 LDAP LoginidAttribute required', () => {
+        cy.visitLDAPSettings();
+
+        // # Try to save LDAP settings with blank Loginid
+        cy.findByTestId('LdapSettings.LoginIdAttributeinput').click().clear();
+        cy.findByText(/save/i).click();
```
Comment (josephbaylon) [inline e2e/cypress/integration/enterprise/ldap/ldap_setting_spec.js:60]:
```suggestion
        cy.findByRole('button', {name: /save/i}).click();
```

File: e2e/cypress/support/ldap_commands.js (hunk 9 above)
Comment (josephbaylon) [inline e2e/cypress/support/ldap_commands.js:12]:
```suggestion
    cy.visit('/admin_console/authentication/ldap');
    cy.get('.admin-console__header').should('be.visible').and('have.text', 'AD/LDAP')
```

**Review by josephbaylon — CHANGES_REQUESTED** [review]
@furqanmlk Looks like some of the suggestions were missed. You can just click `Commit Suggestion` if you don't have further local changes.

File: e2e/cypress/integration/enterprise/ldap/ldap_setting_spec.js [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+
+// Group: @enterprise @ldap
+
+// assumes the CYPRESS_* variables are set
+// assumes that E20 license is uploaded
+// for setup with AWS: Follow the instructions mentioned in the mattermost/platform-private/config/ldap-test-setup.txt file
+
+describe('LDAP settings', () => {
+    beforeEach(() => {
+        // Login as sysadmin
```
Comment (josephbaylon) [inline e2e/cypress/integration/enterprise/ldap/ldap_setting_spec.js:18]:
```suggestion
        // # Login as sysadmin
```

File: e2e/cypress/integration/enterprise/ldap/ldap_setting_spec.js [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+        cy.apiAdminLogin();
+
+        // Check if server has license for LDAP
```
Comment (josephbaylon) [inline e2e/cypress/integration/enterprise/ldap/ldap_setting_spec.js:21]:
```suggestion
        // * Check if server has license for LDAP
```

File: e2e/cypress/integration/enterprise/ldap/ldap_setting_spec.js [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+        cy.apiRequireLicenseForFeature('LDAP');
+    });
+
+    it('MM-T2699 Connection test button - Successful', () => {
+        // # Load AD/LDAP page in system console
+        cy.visitLDAPSettings();
+
+        // click "AD/LDAP Test"
```
Comment (josephbaylon) [inline e2e/cypress/integration/enterprise/ldap/ldap_setting_spec.js:29]:
```suggestion
        // # Click "AD/LDAP Test"
```

File: e2e/cypress/integration/enterprise/ldap/ldap_setting_spec.js [hunk 14]
```diff
@@ -0,0 +1,81 @@
...
+        // # Remove text from Username Attribute
+        cy.findByLabelText(/username attribute:/i).click().clear();
+
+        // # Click Save
+        cy.findByRole('button', {name: /save/i}).click();
+
+        // * Verifying message "AD/LDAP field "Username Attribute" is required."
+        cy.findByText('AD/LDAP field "Username Attribute" is required.').should('be.visible');
+
+        // * Set back to what it was
```
Comment (josephbaylon) [inline e2e/cypress/integration/enterprise/ldap/ldap_setting_spec.js:49]:
```suggestion
        // # Set back to what it was
```

**Review by josephbaylon — APPROVED** [review]
LGTM. Tests passed locally. Thanks @furqanmlk !!

File: e2e/cypress/integration/enterprise/ldap/ldap_setting_spec.js [hunk 15]
```diff
@@ -0,0 +1,81 @@
...
+                username: 'test.two',
+                password: 'Password1',
+            },
+            siteName: 'Mattermost',
+        };
+
+        // # Login as a new LDAP user
+        cy.doLDAPLogin(testSettings);
+
+        // * Verify user is logged in  Successfy
```
Comment (sbishel) [inline e2e/cypress/integration/enterprise/ldap/ldap_setting_spec.js:78]:
```suggestion
        // * Verify user is logged in  Successfully
```

**Review by sbishel — APPROVED** [review]
just a nit.

File: e2e/cypress/integration/enterprise/ldap/ldap_setting_spec.js (hunk 15 above)
Comment (furqanmlk) [reply e2e/cypress/integration/enterprise/ldap/ldap_setting_spec.js]: Thanks:)

**Pull request merged**